Automattic/woocommerce-payments
# Use @wordpress/data in Digital wallets & express payment methods

Fixes https://github.com/Automattic/woocommerce-payments/issues/1728

#### Changes proposed in this Pull Request
- Add the ability to save the following settings in our state store.
![image](https://user-images.githubusercontent.com/572862/118877887-3485c600-b8ac-11eb-89af-0f248ad05081.png)
- Modified `class-wc-rest-payments-settings-controller.php` constructor to allow injection for `Digital_Wallets_Payment_Gateway`, so that it can uses its gateway's built-in enable/disable status.









#### Testing instructions
1. Turn on the group feature flag `_wcpay_feature_grouped_settings` and set it to 1 . Alternatively, [change this line locally](https://github.com/Automattic/woocommerce-payments/blob/14500f9c1d04929a10a56bd3b1164a5d9cdc3134/includes/class-wc-payments-features.php#L22) to return 1.
2. Go to wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments, scroll down and you should see this section:
![image](https://user-images.githubusercontent.com/572862/118878589-f5a44000-b8ac-11eb-94b3-9fa68cbc58d1.png)
3. Uncheck the "Enable 1-click checkouts" checkbox should disable the 3 checkboxes in "Show 1-click checkouts on:".
4. Check the "Enable 1-click checkouts" checkbox and check the 3 checkboxes in "Show 1-click checkouts on:", scroll down and click "Save changes."
5. Refresh the page and you should see the settings saved.

-------------------
